### PR TITLE
[Tutorial - streamlit] - added streamlit neural model mnist script

### PR DESCRIPTION
**Pull request by akisaini**
This PR:

- Contains functionality for resetting the training parameters and running the prediction model.
- Removed the form on the page and using a placeholder instead.
- Added drawable canvas widget to predict hand-drawn digits.
- Refactored code and inverted the canvas background and stroke color.

### CODE REVIEW

File: tutorials/streamlit/MNIST/neural_model.py [hunk 1]
```diff
@@ -1,89 +1,111 @@
...
 num_classes = 10  # total classes (0-9 digits).
 num_features = 784  # data features (img shape: 28*28).
 
-# Initial Training parameters.
+# Initial Training parameters
+training_steps = 500
+display_step = 100
 learning_rate = 0.1
-training_steps = 2000
 batch_size = 256
```
Comment (ktaletsk) [inline tutorials/streamlit/MNIST/neural_model.py:62]: - These parameters leads to only 10% accuracy on the test set
- Keep the same order of variables

File: tutorials/streamlit/MNIST/neural-model.py [hunk 2]
```diff
@@ -145,8 +173,7 @@ def call(self, x, is_training=False):
 # Build neural network model.
 neural_net = NeuralNet()
 # Cross-Entropy Loss.
-# Note that this will apply 'softmax' to the logits.
-
+# Note that this will apply 'softmax' to thwe logits.
```
Comment (ktaletsk) [inline tutorials/streamlit/MNIST/neural-model.py:176]: Typo

File: tutorials/streamlit/MNIST/neural-model.py [hunk 3]
```diff
@@ -1,102 +1,131 @@
 # For Streamlit:
-
+#%%
```
Comment (ktaletsk) [inline tutorials/streamlit/MNIST/neural-model.py:2]: What is this?

File: tutorials/streamlit/MNIST/neural-model.py [hunk 4]
```diff
@@ -1,102 +1,131 @@
...
 import random
-import matplotlib.pyplot as plt
 import numpy as np
 import pandas as pd
 import streamlit as st
 import tensorflow as tf
 from PIL import Image
 from tensorflow import keras
-from keras import Model, layers
+from keras import layers, Model
```
Comment (ktaletsk) [inline tutorials/streamlit/MNIST/neural-model.py:11]: Change the order back to original

File: tutorials/streamlit/MNIST/neural-model.py [hunk 5]
```diff
@@ -239,64 +255,134 @@ def train_model(training_steps, display_step, learning_rate, batch_size):
...
+    #processed_image.show()
+
+    drawn_image_tensor = tf.convert_to_tensor(input_data_final_float, dtype=tf.float32)
+    drawn_image_tensor = tf.expand_dims(drawn_image_tensor, axis=0)
+
+
+    prediction_drawable = neural_net(drawn_image_tensor, is_training=False)
+
+    # Display model prediction on drawable image
+    st.write(f"**_Drawable Prediction:_** {np.argmax(prediction_drawable.numpy())}")
```
Comment (ktaletsk) [inline tutorials/streamlit/MNIST/neural-model.py:388]: Should be `Prediction` instead of `Drawable Prediction`

**Review by ktaletsk — CHANGES_REQUESTED** [review]
Small code cleanup requests

File: tutorials/streamlit/MNIST/neural-model.py [hunk 6]
```diff
@@ -239,64 +254,123 @@ def train_model(training_steps, display_step, learning_rate, batch_size):
...
 st.button("Shuffle Images", on_click=shuffle_images)
+
+
+st.subheader('Try It:')
+
+
+canvas = st_canvas(
+    stroke_width=40, # defaults to 20
+    stroke_color="white",
+    background_color="black",
```
Comment (ktaletsk) [inline tutorials/streamlit/MNIST/neural-model.py:328]: Now that we got it working, we should try to go back to black stroke on white background for visual consistency
```
    stroke_color="black",
    background_color="white",
```

Comment (ktaletsk) [reply tutorials/streamlit/MNIST/neural-model.py:328]: You would use ``ImageOps.invert` from `PIL`: 
https://note.nkmk.me/en/python-pillow-invert/

image

**Review by ktaletsk — CHANGES_REQUESTED** [review]
Some more changes required to validate the user input and improve visual style

**Review by ktaletsk — APPROVED** [review]
